e2e_playwright/st_magic_test.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from playwright.sync_api import Page, expect
+
+
+def test_magic_shows_correct_text(app: Page):
+    expected = [
+        "no block",
+        "This should be printed",
+        "IF",
+        "ELIF",
+        "ELSE",
+        "FOR",
+        "WHILE",
+        "WITH",
+        "TRY",
+        "EXCEPT",
+        "FINALLY",
+        "FUNCTION",
+        "ASYNC FUNCTION",
+        "ASYNC FOR",
+        "ASYNC WITH",
+    ]
+    for i in range(len(expected)):
+        expect(app.get_by_test_id("stMarkdown").nth(i).locator("p")).to_have_text(
```

Comment:
nit: can you check if we even need the `.locator("p"))` here? I believe it might just work with:

```suggestion
        expect(app.get_by_test_id("stMarkdown").nth(i).to_have_text(
```

Reply:
If it doesn't work. maybe try to use `use_inner_text=True` as a paramter